src/js/media-tooltip.ts

```diff
@@ -0,0 +1,239 @@
+      top: 50%;
+      left: 100%;
+      border-width: 6px 0 6px 5px;
+      border-color: transparent transparent transparent var(--_tooltip-background);
+      transform: translate(0, -50%);
+    }
+    
+    :host([position="none"]) #arrow {
+      display: none;
+    }
+
+  </style>
+  <slot></slot>
+  <div id="arrow"></div>
+`;
+
+/**
+ * @extends {HTMLElement}
+ *
+ * @attr {('top'|'right'|'bottom'|'left'|'none')} position - The position of the tooltip, defaults to "top"
+ * @attr {string} container - CSS selector for the containing element (one of it's parents) that should constrain the tooltips horizontal position.
+ *
+ * @cssproperty --media-primary-color - Default color of text.
+ * @cssproperty --media-secondary-color - Default color of tooltip background.
+ * @cssproperty --media-text-color - `color` of tooltip text.
+ *
+ * @cssproperty --media-font - `font` shorthand property.
+ * @cssproperty --media-font-weight - `font-weight` property.
+ * @cssproperty --media-font-family - `font-family` property.
+ * @cssproperty --media-font-size - `font-size` property.
+ * @cssproperty --media-text-content-height - `line-height` of button text.
+ *
+ * @cssproperty --media-tooltip-background - `background` color
+ * @cssproperty --media-tooltip-display - `display` of tooltip
+ * @cssproperty --media-tooltip-z-index - `z-index` of tooltip
+ * @cssproperty --media-tooltip-padding - `padding` of tooltip
+ * @cssproperty --media-tooltip-border-radius - `border-radius` of tooltip
+ * @cssproperty --media-tooltip-filter - `filter` property of tooltip, for drop-shadow
+ * @cssproperty --media-tooltip-white-space - `white-space` property of tooltip
+ * @cssproperty --media-tooltip-arrow-display - `display` property of tooltip arrow
+ */
+class MediaTooltip extends globalThis.HTMLElement {
+  static get observedAttributes(): string[] {
+    return [Attributes.POSITION, Attributes.CONTAINER];
+  }
+
+  arrowEl: HTMLElement;
+
+  constructor() {
+    super();
+
+    if (!this.shadowRoot) {
+      // Set up the Shadow DOM if not using Declarative Shadow DOM.
+      this.attachShadow({ mode: 'open' });
+      this.shadowRoot.appendChild(template.content.cloneNode(true));
+    }
+
+    this.arrowEl = this.shadowRoot.querySelector('#arrow');
+  }
+
+  // Adjusts tooltip position relative to the closest specified container
+  // such that it doesn't spill out of the left or right sides. Only applies
+  // to 'top' and 'bottom' positioned tooltips.
+  updateXOffset = () => {
+    const position = this.position;
+
+    // we don't offset against tooltips coming out of left and right sides
+    if (position === 'left' || position === 'right') {
+      // could have been offset before switching to a new position
+      this.style.removeProperty('--media-tooltip-offset-x');
+      return;
+    }
+
+    const containingSelector = this.container;
+    const containingEl = closestComposedNode(this, containingSelector);
+    if (!containingEl) return;
```

Comment:
I think it'd be good to default this to `media-controller` specifically. The main thing you don't want is the tooltip going outside of the player, especially in full screen mode, and the media-controller best represents that most of the time. The control-bar is a little too optional to rely on I think, and the control-bar left/right margin can need to be zero for some designs, as opposed to always relying on the control bar providing the margin between the tooltip and the edge of the screen.

But so with that, I think we'd want some kind of default margin between the tooltip and the edge of the media controller or supplied container. Like `--media-tooltip-container-margin`. How does that sound?

Reply:
That makes sense. Currently `media-xyz-button`'s are defining the containing element, but we could switch it to a default (media-controller) that can be optionally overridden and manually adjusted with a margin.

Reply:
Have pushed up some changes for this. Defaulting to `media-controller` and with an optional margin css var to pad it out.

Reply:
sorry, this will need additional changes. 
we've used the `bounds` attribute for several elements already (time-range, menu). 
can we align on that? 

see https://github.com/muxinc/media-chrome/blob/main/src/js/experimental/media-chrome-selectmenu.ts#L346-L362

Reply:
@luwes, do you think someone might want to set different bounds for the tooltips compared to what the `bounds` attribute is already added to? e.g. someone wants all the menus to contain within one thing, but the tooltips to be contained by something else (maybe something wrapping just the buttons). I'm happy to piggy back on the `bounds` attribute if that doesn't sound like something that's needed. I also prefer the consistency if we did use it if there's no issue with the above.

Reply:
@AdamJaggard they could still do that I think, the `bounds` attribute is set on the timerange element or the menu element or on the tooltip element. 
https://github.com/muxinc/media-chrome/blob/main/src/js/media-chrome-menu.ts#L956-L960

Reply:
Hey @luwes, have updated this to `bounds`, let me know if I've missed anything matching current implementation. I didn't re-use the `getBoundsElement` function as it falls back to the immediate parent (or itself) as the bounds element if it can't find anything and this breaks tooltip assumptions.